R/utils.R

```diff
@@ -45,3 +45,76 @@ is_silent <- function(segments) {
 silent <- function(){
   "[...]"
 }
+
+#' Parse time if it is a string
+#'
+#' @param time A time string in the "HH:MM(:SS)( AM/PM)", format.
+#' @param format A character vector of formats to try.
+#'
+#' @return A POSIXct value representing the time.
+#'
+parse_event_time <- function(time, format = c("R", "T")) {
+  if (!inherits(time, c("POSIXct", "character"))) {
+    stop("Invalid time format. Should be either a valid HH:MM(:SS)( AM/PM) ",
+         "character or already in POSIXct format.")
+  }
+
+  if (inherits(time, "POSIXct")) {
+    time
+  } else {
+    lubridate::parse_date_time(time, format)
+  }
+}
```

Comment:
No, parse_date_time can receive only characters since all other data types would be managed before reaching it. And if the format is not right, it will return an NA, but it won't raise an error. The NA will be managed on a per-situation basis by the other functions